tests/csv_upload_tests.py

```diff
@@ -128,218 +141,123 @@ def test_import_csv_in_new(self):
+        conf["POSTGRES_PASSWORD"] = POSTGRES
         try:
             response = self.get_resp(url, data=form_data)
-            message = "No password supplied for PostgreSQL"
+            message = "{0} imported into database {1}".format(table_name, db_name)
+            print(response)  # show error message in build
             assert message in response
         finally:
             os.remove(filename)
+            url = (
+                "postgresql://"
+                + conf["POSTGRES_USERNAME"]
+                + ":"
+                + conf["POSTGRES_PASSWORD"]
+                + "@localhost/"
+                + db_name
+            )
+            if sqlalchemy_utils.database_exists(url):
+                sqlalchemy_utils.drop_database(url)
 
-    def test_postgres_no_username_supplied(self):
-        url = "/csvimporter/csvtodatabase/add"
-        filename = "postgres_no_username.csv"
-        db_name = "csv_into_new_postgres_no_username"
-        table_name = "postgres_no_username"
-        form_data = self.get_full_data(
-            filename, NEW_DATABASE_ID, db_name, table_name, database_flavor=POSTGRES
+    def test_allow_csv_upload_databases(self):
+        db_id = self.get_existing_db_id()
+        assert db_id in [
+            database.id for database in self.importer.allow_csv_upload_databases()
+        ]
+
+    def test_clean_filename(self):
+        original_filename = "foo,+.bar"
+        filename = self.importer._clean_filename(original_filename, "CSV")
+        print(filename)
+        assert filename == "foo.bar"
+        filename = self.importer._clean_filename(filename, "CSV")
+        assert filename == "foo.bar"
+
+    def test_clean_filename_None(self):
+        purpose = "CSV"
+        error_message = "No name is received for {0}".format(purpose)
+        with self.assertRaisesRegex(NameNotAllowed, error_message):
+            self.importer._clean_filename(None, purpose)
+
+    def test_clean_filename_empty(self):
+        filename = "#,'\""
+        purpose = "CSV"
+        error_message = "Name {0} is not allowed for {1}".format(filename, purpose)
+        with self.assertRaisesRegex(NameNotAllowed, error_message):
+            self.importer._clean_filename(filename, purpose)
+
+    def test_convert_database_id(self):
+        database_id = "1"
+        converted_id = self.importer._convert_database_id(database_id)
+        assert converted_id == 1
+
+    def test_convert_database_id_failed(self):
+        database_id = "id"
+        error_message = (
+            "Possible tampering detected, non-numeral character in database-id"
         )
-        conf["POSTGRES_USERNAME"] = ""
-        conf["POSTGRES_PASSWORD"] = POSTGRES
+        with self.assertRaisesRegex(ValueError, error_message):
+            self.importer._convert_database_id(database_id)
+
+    def test_check_table_name(self):
+        table_name = "myNewTableName"
+        assert not self.importer._check_table_name(table_name)
```

Comment:
not sure if this belongs here but maybe rename this method? something more clear like "check_if_table_name_already_exists"?
and that method returns false or throws an exception, why not true/false and handle this outside the method?